In [2]:
import pandas as pd

In [4]:
feature_events = pd.read_csv('/Users/vaishnav/Documents/Project/ai-product-impact-analysis/data/raw/feature_events.txt')
subscriptions = pd.read_csv('/Users/vaishnav/Documents/Project/ai-product-impact-analysis/data/raw/subscriptions.txt')
support_tickets = pd.read_csv('/Users/vaishnav/Documents/Project/ai-product-impact-analysis/data/raw/support_tickets.txt')
users = pd.read_csv('/Users/vaishnav/Documents/Project/ai-product-impact-analysis/data/raw/users.txt')

In [5]:
feature_events.head()

,event_id,user_id,event_name,feature_name,event_date,session_id,device_type,duration_seconds,success
0,1,1,project_created,Core Product,2024-10-19 20:11:18,ce465118,Desktop,44,True
1,2,1,project_created,Core Product,2024-10-26 06:40:19,8b89e63c,Desktop,8,True
2,3,1,comment_added,Core Product,2024-11-01 16:41:15,6becbb25,Desktop,15,True
3,4,1,team_invited,Core Product,2024-11-01 16:41:15,6becbb25,Desktop,16,True
4,5,1,task_created,Core Product,2024-10-11 21:48:46,f3ba1742,Mobile,58,True


total rows = 6467322

event_id is unique identifier

feature = 5 ['Core Product', 'SmartAssist', 'smart_assist', 'Smart Assist', 'SMARTASSIST']

device type = 3 ['Desktop', 'Mobile', 'Tablet']

In [11]:
feature_events.nunique()

event_id            6467322
user_id               42777
event_name               13
feature_name              5
event_date           978404
session_id          1029312
device_type               3
duration_seconds        391
success                   2
dtype: int64

In [18]:
feature_events['event_name'].unique()

<StringArray>
[       'project_created',          'comment_added',           'team_invited',
           'task_created',          'file_uploaded',       'settings_updated',
       'dashboard_viewed',       'report_generated', 'ai_suggestion_rejected',
        'ai_help_clicked',  'ai_template_generated', 'ai_suggestion_accepted',
    'ai_prompt_submitted']
Length: 13, dtype: str

In [19]:
subscriptions.head()

,account_id,company_name,plan_tier,mrr,status,start_date,churn_date,upgrade_date,company_size,industry,acquisition_channel,sales_rep,contract_length_months,discount_percentage,region,payment_method
0,1,"Rodriguez, Figueroa and Sanchez",Pro,100.25,active,2024-09-07,NaN,NaN,11-50,Financial Services,Direct Sales,Alex Johnson,1,0.0,Europe,ACH
1,2,Doyle Ltd,Starter,29.60,active,2024-08-15,NaN,NaN,1-10,Media & Entertainment,Organic Search,None (Self-Serve),1,0.0,Europe,Credit Card
2,3,"Mcclain, Miller and Henderson",Starter,34.56,active,2024-08-22,NaN,NaN,1-10,Professional Services,Organic Search,None (Self-Serve),12,12.6,Latin America,Credit Card
3,4,Davis and Sons,Starter,36.60,active,2024-08-08,NaN,NaN,11-50,Non-Profit,Organic Search,None (Self-Serve),1,0.0,North America,Credit Card
4,5,"Guzman, Hoffman and Baldwin",Enterprise,1066.80,active,2024-09-19,NaN,NaN,1000+,Education,Direct Sales,Alex Johnson,12,13.3,Europe,Credit Card


row count = 2500

account_id is unique identifier

plan_tier = 3 ['Pro', 'Starter', 'Enterprise']

status = ['active', 'churned']

company_size = 5 ['11-50', '1-10', '1000+', '51-200', '201-1000']

region = 5 ['Europe', 'Latin America', 'North America', 'Asia Pacific']

In [21]:
subscriptions.nunique()

account_id                2500
company_name              2371
plan_tier                    3
mrr                       2101
status                       2
start_date                  62
churn_date                  59
upgrade_date                 0
company_size                 5
industry                    10
acquisition_channel          7
sales_rep                    8
contract_length_months       3
discount_percentage        190
region                       4
payment_method               4
dtype: int64

In [33]:
subscriptions['contract_length_months'].unique()

array([ 1, 12, 24])

In [34]:
support_tickets.head()

,ticket_id,account_id,created_date,resolved_date,category,priority,channel,assigned_to,ticket_description,resolution_time_hours,first_response_time_hours,sentiment_score,csat_score,escalated,month
0,1,1,2024-11-19 13:30:28,2024-11-19 16:09:00.046458,Billing Question,Critical,In-App Chat,Agent_C,Protect serious under imagine adult degree who...,2.64,1.17,-0.04,2,False,2024-11
1,2,2,2024-10-15 17:59:05,2024-10-16 00:09:10.060885,SmartAssist,Medium,Email,Agent_B,Series seek world audience page local medical....,6.17,4.78,-0.19,1,True,2024-10
2,3,2,2024-10-28 11:32:37,2024-10-28 18:46:02.711880,SmartAssist,Critical,Email,Agent_B,Mother go while administration have. automatio...,7.22,1.63,-0.29,3,False,2024-10
3,4,2,2024-09-23 04:10:32,2024-09-23 10:09:06.523487,SmartAssist,Low,Help Center,Agent_E,Force know discuss hair. AI feature confusing ...,5.98,6.70,0.24,3,False,2024-09
4,5,3,2024-12-09 02:42:28,2024-12-09 07:33:42.290302,Bug Report,Low,In-App Chat,Agent_D,Discover stuff simple would area outside. Buil...,4.85,8.47,-0.17,3,False,2024-12


row count = 5859

priority = 4 ['Critical', 'Medium', 'Low', 'High']

In [50]:
support_tickets.nunique()

ticket_id                    5859
account_id                   2104
created_date                 5859
resolved_date                5859
category                        8
priority                        4
channel                         4
assigned_to                     6
ticket_description           5859
resolution_time_hours        1018
first_response_time_hours    1524
sentiment_score               131
csat_score                      5
escalated                       2
month                           5
dtype: int64

In [52]:
support_tickets['priority'].unique()

<StringArray>
['Critical', 'Medium', 'Low', 'High']
Length: 4, dtype: str

In [53]:
users.head()

,user_id,account_id,email,full_name,role,job_title,department,license_type,signup_date,last_login_date,login_count,feature_adoption_score,is_active
0,1,1,gomezlisa@example.net,Christopher Tate,member,Marketing Manager,Finance,Full,2024-10-05,2024-12-29,123,61,True
1,2,1,martinezmitchell@example.org,Daniel Smith,member,Designer,Product,Full,2024-09-28,2024-12-31,18,80,True
2,3,1,erin53@example.org,Lori Stokes,member,Marketing Manager,Engineering,Limited,2024-10-06,2024-12-20,150,51,True
3,4,1,jimsnyder@example.com,Johnathan Padilla,member,Project Manager,Operations,Full,2024-09-09,2024-12-17,170,65,True
4,5,1,adrianjoyce@example.net,Sergio Carroll,member,Data Analyst,Sales,Full,2024-09-19,2024-12-27,111,45,True


row count = 42777

role = 3 ['member', 'admin', 'guest']

license_type = 3 ['Full', 'Limited', 'Viewer']

users = 1 [True]

In [56]:
users.nunique()

user_id                   42777
account_id                 2500
email                     39550
full_name                 35197
role                          3
job_title                    19
department                    8
license_type                  3
signup_date                  92
last_login_date              15
login_count                 308
feature_adoption_score       86
is_active                     1
dtype: int64

In [59]:
users['is_active'].unique()

array([ True])

In [66]:
subscriptions.isnull().sum()

account_id                   0
company_name                 0
plan_tier                    0
mrr                          0
status                       0
start_date                   0
churn_date                2283
upgrade_date              2500
company_size                 0
industry                     0
acquisition_channel          0
sales_rep                    0
contract_length_months       0
discount_percentage          0
region                       0
payment_method               0
dtype: int64

In [108]:
subscriptions['churn_date'].max()

Timestamp('2024-12-31 00:00:00')

In [99]:
subscriptions.info()

<class 'pandas.DataFrame'>
RangeIndex: 2500 entries, 0 to 2499
Data columns (total 16 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   account_id              2500 non-null   int64  
 1   company_name            2500 non-null   str    
 2   plan_tier               2500 non-null   str    
 3   mrr                     2500 non-null   float64
 4   status                  2500 non-null   str    
 5   start_date              2500 non-null   str    
 6   churn_date              217 non-null    str    
 7   upgrade_date            0 non-null      float64
 8   company_size            2500 non-null   str    
 9   industry                2500 non-null   str    
 10  acquisition_channel     2500 non-null   str    
 11  sales_rep               2500 non-null   str    
 12  contract_length_months  2500 non-null   int64  
 13  discount_percentage     2500 non-null   float64
 14  region                  2500 non-null   str    
 15

In [115]:
subscriptions.churn_date=pd.to_datetime(subscriptions['churn_date'])
subscriptions.start_date=pd.to_datetime(subscriptions['start_date'])

In [116]:
subscriptions.info()

<class 'pandas.DataFrame'>
RangeIndex: 2500 entries, 0 to 2499
Data columns (total 16 columns):
 #   Column                  Non-Null Count  Dtype         
---  ------                  --------------  -----         
 0   account_id              2500 non-null   int64         
 1   company_name            2500 non-null   str           
 2   plan_tier               2500 non-null   str           
 3   mrr                     2500 non-null   float64       
 4   status                  2500 non-null   str           
 5   start_date              2500 non-null   datetime64[us]
 6   churn_date              217 non-null    datetime64[us]
 7   upgrade_date            0 non-null      float64       
 8   company_size            2500 non-null   str           
 9   industry                2500 non-null   str           
 10  acquisition_channel     2500 non-null   str           
 11  sales_rep               2500 non-null   str           
 12  contract_length_months  2500 non-null   int64         
 13 

In [124]:
print("Subscriptions:", subscriptions["start_date"].dt.date.min(), "to ", subscriptions["start_date"].dt.date.max())

Subscriptions: 2024-08-01 to  2024-10-01


In [128]:
feature_events['feature_name'].unique()

<StringArray>
['Core Product', 'SmartAssist', 'smart_assist', 'Smart Assist', 'SMARTASSIST']
Length: 5, dtype: str

In [136]:
feature_events['feature_name_clean']=(feature_events['feature_name'].str.lower().str.replace('_','').str.replace(" ",''))

In [138]:
feature_events['feature_name_clean'].value_counts()

feature_name_clean
coreproduct    6433767
smartassist      33555
Name: count, dtype: int64